### DISCUSSION TITLE
Live theme change?

### DISCUSSION THREAD

**Discussion by MyGitHubBlueberry** [nix-community/stylix#371] [Q&A]
This project looks really cool.
In case I use feature to generate colors from an image and then change it to another one, I'm curious if it will pick-up color change live without need to `nixos-rebuild switch`, or `home-manager switch` and/or reopening apps.

**Comment by trueNAHO** [comment]
> This project looks really cool.

I'm glad you like the project.

> In case I use feature to generate colors from an image and then change it to another one, I'm curious if it will pick-up color change live without need to `nixos-rebuild switch`, or `home-manager switch` and/or reopening apps.

This sounds like a neat feature. Actually, https://github.com/danth/stylix/pull/102 was meant to extend wallpaper capabilities, which would somewhat allow such a functionality. Unfortunately, it has been stale for a while.

**Comment by danth** [comment]
`nixos-rebuild switch` / `home-manager switch` will always be necessary since all the theming is done as part of building the configuration.

Whether or not an app needs to be re-opened can vary - some will update instantly, some won't reload their config until they're restarted. Live reloading needs to be implemented by the original author of the app before Stylix can support it.

**Comment by ghost** [comment]
I wonder if there's a way to accomplish this using PyWal to generate the colors and just have the generated colors in the .cache/wal directory used as stylix's colors.

**Reply by trueNAHO** [reply]
I am not familiar with [Pywal](https://github.com/dylanaraps/pywal). Could you explain how it updates colorschemes in real-time? If it uses a daemon that regenerates the configurations at run time when the color palette changes, it might conflict with the imutability nature of Nix.

**Reply by ghost** [reply]
Well, it generates colors from whatever wallpaper is currently set at run-time and spits out various files of the colors generated in different formats (css, json, and some app specific configs like hyprland, sway, and rofi).  I just import the relevant ones into my home-manager configs for the relevant apps and have their configs set up appropriately to reference them.  I also have a script set up so that when I change wallpapers, pywal gets run to update the color schemes which then updates the UI.

I am likely missing a lot in that explanation, I am not an expert by any means.

my config is here, if you want to see what I'm doing: https://github.com/angerZen/flakes

**Reply by trueNAHO** [reply]
Related: https://github.com/danth/stylix/issues/99
